econml/tests/test_dmliv.py

```diff
@@ -187,3 +187,54 @@ def true_fn(X):
                 ate_lb, ate_ub = est.effect_interval()
                 np.testing.assert_array_less(ate_lb, true_ate)
                 np.testing.assert_array_less(true_ate, ate_ub)
+
+    def test_groups(self):
+        n = 500
+        d_w = 10
+        d_x = 3
+        y = np.random.normal(size=(n,))
+        W = np.random.normal(size=(n, d_w))
+        X = np.random.normal(size=(n, d_x))
+        T = np.random.choice(["a", "b"], size=(n,))
+        Z = np.random.choice(["c", "d"], size=(n,))
+        groups = [i // 3 for i in range(n)]
+
+        est_list = [
+            OrthoIV(
+                projection=False,
+                discrete_treatment=True,
+                discrete_instrument=True,
+                model_y_xw=LinearRegression(),
+                model_t_xw=LogisticRegression(),
+                model_z_xw=LogisticRegression(),
+            ),
+            OrthoIV(
+                projection=True,
+                discrete_treatment=True,
+                discrete_instrument=True,
+                model_y_xw=LinearRegression(),
+                model_t_xw=LogisticRegression(),
+                model_t_xwz=LogisticRegression(),
+            ),
+            DMLIV(
+                model_final=LinearRegression(fit_intercept=False),
+                discrete_treatment=True,
+                discrete_instrument=True,
+                model_y_xw=LinearRegression(),
+                model_t_xw=LogisticRegression(),
+                model_t_xwz=LogisticRegression(),
+            ),
+            NonParamDMLIV(
+                model_final=RandomForestRegressor(),
+                discrete_treatment=True,
+                discrete_instrument=True,
+                model_y_xw=LinearRegression(),
+                model_t_xw=LogisticRegression(),
+                model_t_xwz=LogisticRegression(),
+            ),
+        ]
+
+        for est in est_list:
+            est.fit(y, T, Z=Z, X=X, W=W, groups=groups)
```

Comment:
Is there a way to make sure the groups are actually being used here? To avoid problems like when groups is accidentally left as None in the call to super().score() instead of threaded through from the args.